isis/src/mro/apps/marcical/main.cpp

```diff
@@ -364,56 +364,38 @@ void IsisMain() {
-    // so we *must* loop through the entire file.
-  for(int i = 0 ; i < csv.rows() ; i++) {
-      CSVReader::CSVAxis row = csv.getRow(i);
-      // The productId in the file is encapsulated by double quotes.
-      QString fileProdId = row[0];
-      //This is garbage code, but my compiler isn't allowing me to escape the double quotes
-      int prodIdLastIndex = fileProdId.size() - 1 ;
-      fileProdId.remove(prodIdLastIndex,1);
-      fileProdId.remove(0,1);
-      // Now, compare product ids from the input image and from the calibration table.
-      if(fileProdId == prodId ) {
-        if((row.dim1() - 1) != 2) {
-          QString msg = "This appears to be a malformed calibration file."; 
-                  msg += " There are not enough columns in the CSV";
-                  msg += " file to perform the exposure time correction.";
-          throw IException(IException::User, msg, _FILEINFO_);
-        }
-        // Build the two vectors, exptime and frame. We'll relate those to each other
-        // back in main(). Remember that a productID may have multiple entries in the table.
-        frameseq.push_back(toInt(row[1]));
-        exptime.push_back(toDouble(row[2]));
-      }
-  }
 
-  if (flipped && exptime.size() > 0) {
-    reverse(frameseq.begin(),frameseq.end());
-    reverse(exptime.begin(),exptime.end());
+  // Get the exposure duration(s) and coorisponding frame number(s) (zero based) from the label
+  PvlGroup inst = icube.group("Instrument");
+
+  if (!inst.hasKeyword("VariableExposureDuration") || !inst.hasKeyword("FrameNumber")) {
+    QString msg = "The instrument keywords VariableExposureDuration and FrameNumber"
+                  "must exist to calibrate this MARCI file";
+    throw IException(IException::User, msg, _FILEINFO_);
   }
 
-  if (exptime.size() == 0) {
-    PvlGroup missing("NoExposureTimeDataFound");
-    missing.addKeyword(PvlKeyword("FileNotFoundInVarexpFile", prodId), Pvl::Replace);
-    Application::Log(missing);
+  // The previous version of marcical read the first exposure duration and frame number from
+  // the label and the other durations/frames (if any) from the mission's varexp.tab file.
+  // In order to minimize the changes to this code while modifying it to use the keywords instead of
+  // reading the varexp.tab file directly, the duration/frame where the frame is zero are dropped from 
+  // the array when converting them from the keywords. This calibration code should be 
+  // able to be simplified significantly by not removing the frameNumber=0 exposure/frame and
+  // removing the first frame ifs in the calibration code below.
+  PvlKeyword expTimesKey = inst["VariableExposureDuration"];
+  PvlKeyword frameNumbersKey = inst["FrameNumber"];
+  for (int i=0; i<expTimesKey.size(); i++) {
+    if (toInt(frameNumbersKey[i]) != 0) {
+      exptime.push_back(toDouble(expTimesKey[i]));
+      frameseq.push_back(toInt(frameNumbersKey[i]));
+    }
```

Comment:
I think you can remove this by just removing the else part of the if/else statements later on lines 438 and 461.